tests/intersection.egg

```diff
@@ -27,8 +27,8 @@
 (let t2p (f (f b2)))
 (union t2 t2p)
 
-(union (intersect a1 a2) a3)
-(union (intersect b1 b2) b3)
+(set (intersect a1 a2) a3)
```

Comment:
this could just be `union`?

Reply:
Are we disallowing `set` instead of `union` in a follow up PR?